src/sqlfluff/core/templaters/dbt.py

```diff
@@ -204,6 +208,41 @@ def process(self, *, fname, in_str=None, config=None):
         except SQLTemplaterError as e:
             return None, [e]
 
+    @contextmanager
+    def _tests_as_models(self):

```

Comment:
@alanmcruickshank @dmateusp I've come up with a way to handle dbt tests, which I'm curious for your thoughts on. It feels really wrong but also works? 😂

By moving the tests into the models directory and removing them from the dbt project config they get templated as models, which prevents them from being wrapped in a CTE which leads to weird linting issues. This function moves the models, patches the dbt project config, and then puts everything back once done.

Reply:
Yeah this does feel kind of gross - but maybe it's worth it as a short term patch. I'm going to have a look through the rest of what you've done and come back to this.

Reply:
Another option which could be less gross - we're already handling potentially wrapped queries. Why don't we just unwrap them in the templater? That's the effect of making them models instead of tests right? Rather than going through all this I think it would be much simpler to just strip the ends of the query. I might have a try and see how easy that would be,

Reply:
I like this option a lot! Feel free.